src/components/domain/project/hooks/useProjectCardClick.ts

```diff
@@ -0,0 +1,37 @@
+import { useRouter } from 'next/navigation';
+import useMessageBox from '@/hooks/useMessageBox';
+import {
+  PROJECT_CARD_VARIANT,
+  type ProjectSummaryCardVariant,
+} from '@/models/project/projectModels';
+
+const useProjectCardClick = (
+  variant: ProjectSummaryCardVariant,
+  projectId: number,
+  userId?: string,
+) => {
+  const router = useRouter();
+  const { showConfirmMessageBox } = useMessageBox();
+
+  const routeMap: Record<ProjectSummaryCardVariant, string> = {
+    [PROJECT_CARD_VARIANT.SEARCH_RESULT]: `/project/${projectId}`,
+    [PROJECT_CARD_VARIANT.ADMIN]: `/project/${projectId}`,
+    [PROJECT_CARD_VARIANT.LINK_PREVIEW]: `/project/${projectId}`,
+    [PROJECT_CARD_VARIANT.MY_PROFILE]: `/project/my/${projectId}`,
+    [PROJECT_CARD_VARIANT.OTHER_PROFILE]: userId ? `/project/user/${userId}/${projectId}` : '',
+  };
+
+  const route = routeMap[variant] || '';
```

Comment:
위와 같이 수정한다면 여기는 이렇게 수정할 수 있을 것 같네요!

```suggestion
  const route = getRoute();
 ```

Reply:
eeafd4719b9ed978bc9e5cdae34a27395e52974d